<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:macro name="joint_interface" params="name">
    <joint name="${name}">
        <command_interface name="position"/>
        <state_interface name="position"/>
    </joint>
  </xacro:macro>

  <ros2_control name="ElkapodLegsSystemHardware" type="system">
      <hardware>
          <plugin>elkapod_description/ElkapodLegsSystemHardware</plugin>
          <param name="uart_device">/dev/ttyAMA1</param>
          <param name="uart_baudrate">1152000</param>
          <param name="uart_timeout_ms">100</param>

          <param name="spi_bus">0</param>
          <param name="spi_cs">0</param>
          <param name="spi_mode">2</param>
          <param name="spi_speed_hz">1000000</param>
      </hardware> 

      <xacro:joint_interface name="leg1_J1"/>
      <xacro:joint_interface name="leg1_J2"/>
      <xacro:joint_interface name="leg1_J3"/>
      
      <xacro:joint_interface name="leg3_J1"/>
      <xacro:joint_interface name="leg3_J2"/>
      <xacro:joint_interface name="leg3_J3"/>
      
      <xacro:joint_interface name="leg5_J1"/>
      <xacro:joint_interface name="leg5_J2"/>
      <xacro:joint_interface name="leg5_J3"/>
      
      
      <xacro:joint_interface name="leg2_J1"/>
      <xacro:joint_interface name="leg2_J2"/>
      <xacro:joint_interface name="leg2_J3"/>
      
      <xacro:joint_interface name="leg4_J1"/>
      <xacro:joint_interface name="leg4_J2"/>
      <xacro:joint_interface name="leg4_J3"/>
      
      <xacro:joint_interface name="leg6_J1"/>
      <xacro:joint_interface name="leg6_J2"/>
      <xacro:joint_interface name="leg6_J3"/>

      <!-- Sensors -->
      <sensor name="imu_sensor" type="imu">
        <param name="frame_id">imu_link</param>
        <param name="update_rate">50.0</param>
        <state_interface name="orientation.x"/>
        <state_interface name="orientation.y"/>
        <state_interface name="orientation.z"/>
        <state_interface name="orientation.w"/>
        <state_interface name="angular_velocity.x"/>
        <state_interface name="angular_velocity.y"/>
        <state_interface name="angular_velocity.z"/>
        <state_interface name="linear_acceleration.x"/>
        <state_interface name="linear_acceleration.y"/>
        <state_interface name="linear_acceleration.z"/>
      </sensor>

  </ros2_control>

</robot>